### PR TITLE
Fixing broken links in Libretime interface + website 2.1

### LINKED ISSUE #333: (Security) How to change default passwords
Looking through the docs, I don't see clear guidance on altering default passwords.  If you have anything else to add, please feel free.  I plan to include icecast and ldap as I look into them and maybe others as I find them.  I will update this first post with changes so this should be consolidated regardless of future post comments till this can be vetted and added to the docs.  Ideal would be a hardening script where these can be updated by command option to something unique.

**[Icecast Admin]** Needs Work - Solution Provided
Default is admin:hackme   It's very important to change as someone can kill your source, move your listeners, update your metadata and list your clients.  Appears to be through the web gui BUT when changed doesn't appear to apply on backend as default admin:hackme works regardless of what is defined in web gui.  To alter your admin username and(or) password:
Edit the following file:
`/etc/icecast2/icecast.xml`
Replace the admin and hackme field below with your own username and password.
```
    <authentication>
        ...
        <!-- Admin logs in with the username given below -->
        <admin-user>admin</admin-user>
        <admin-password>hackme</admin-password>
        ...
    </authentication>
```
Restart your icecast2 service
`service icecast2 restart`
Note: There may be a bug with the GUI as it does not show the new username and password that you setup in the icecast.xml file but this should otherwise work.

**[Icecast Mount Password]** Needs Work
Default is hackme  I know this would be the password that liquidsoap passes to icecast to be able to play for the primary mount but I don't know if this is only on localhost or available on the public IP.  Help appreciated to complete this one.

Edit the following file: 
`/etc/icecast2/icecast.xml`
Replace the hackme field below with your own password.
```
<icecast>
    ...
    <authentication>
        <source-password>hackme</source-password>
        ...
    </authentication>
    ...
</icecast>
```
Note: There may be a bug with the GUI as it does not show the updated mountpoint password that you setup in the icecast.xml file.  Liquidsoap would also have to be updated to be able to mount the stream.

**[LDAP]** Needs Work
Default is hackme  Haven't researched if the interface for this is public facing or just localhost.
FreeIPA/LDAP integration doesn't work out of the box as it needs manual configuration specific to your IPA domain. As such it's deactivated per default and won't do anything.

All in all it is a rather enterprisey feature. It has [docs](http://libretime.org/freeipa/) that detail the setup. I'm assuming those might need more thorough testing since they are mostly grabbed from my production environment.

**[LibreTime]** Completed
The default account/password is admin admin.  Use this to login to the website for the first time. yourlibretimeserverfqdn/login  Once logged on, select the "admin" link at the upper right, change the passwords and then select save.

**[postgresql]** Completed
`sudo -u postgres psql -d postgres -tAc "ALTER USER airtime WITH ENCRYPTED PASSWORD 'newpassword';"`
Then update the following files with the new password:
database section:
/etc/airtime/airtime.conf
database driver section:
/usr/share/airtime/php/airtime_mvc/build/build.properties
 <datasource id="airtime"> dsn section:
/usr/share/airtime/php/airtime_mvc/build/runtime-conf.xml

**[rabbitmq]** Completed
`sudo rabbitmqctl change_password airtime newpassword`
Then update the following file with the new password:
/etc/airtime/airtime.conf

### LINKED ISSUE #1095: No manual pages from interface links
Dashboard - Tracks

You haven't added any tracks
Click 'Upload' to add some now.
Learn about tracks -> libretime.org/manual/tracks -> 404 Tape Page not found :(

Same at Playlists when empty
Same at Smart Blocks when empty
Same at Webstreams when empty
Same at Podcasts when empty

### PR DESCRIPTION
**Pull request by zklosko**
This PR closes #1095. Also closes #333. While I've been diving in the code, I've decided to spruce up the website a bit as well and finish up #376. Github Pages isn't behaving on my end, so to preview you'll need to [set up a Jekyll dev environment](https://jekyllrb.com/docs/).

When Jekyll is set up, cd to the /docs directory of the repo (clone it locally first) and run `bundler install` and then `jekyll serve`.

I've also removed the ugly red logout button in the lower left corner. We already have a link to log out in the upper right, where such links traditionally are, and removing the large red button makes the interface look more professional. I've checked the functionality of Libretime in my test environment and everything still works as intended.

-Zack

P.S. I'm also adding backup instructions and scripts as per #190

### CODE REVIEW

**Comment by paddatrapper** [comment]
I think the changes for #1099 and #1043 should be separate PRs

**Comment by zklosko** [comment]
@paddatrapper @Robbt I'm running into an issue with building the site with absolute links (which allow us to organize the site better). If we publish when I'm finished, we might run into the issue of having a broken site served from https://libretime.github.io/libretime.

Ex. some of the URLs would point to /install in dev might get attached to libretime.github.io/install in production.

Do you know how our domain is connected to our repo? Would it serve /install as libretime.org/install?

**Comment by zklosko** [comment]
Oops. I hit the wrong button.

File: docs/Gemfile.lock [hunk 1]
```diff
@@ -62,7 +62,7 @@ PLATFORMS
 
 DEPENDENCIES
   jekyll (~> 4.1)
-  kramdown (~> 2.2, >= 2.2.1)
+  kramdown (>= 2.3.0)
```
Comment (paddatrapper) [inline docs/Gemfile.lock:65]: This change has been committed already

**Comment by zklosko** [comment]
@paddatrapper @Robbt Requesting review

**Comment by paddatrapper** [comment]
# Set up
```
$ bundle exec jekyll serve
Could not locate Gemfile or .bundle/ directory
```

Had to rename `docs/gemfile` to `docs/Gemfile`.

# Notes
## Home page

- [ ] The sub-title ('Everything you need to get your terrestrial/on-line...') is hard to read over the background. Maybe the font should be thicker?
- [ ]  'no linger actively maintained' - does this need to be bold? I think the important part here is that LibreTime is open source, not that Airtime is not maintained
- [ ]  All the image links (library, stats, schedule, etc) on the home page return 404 page not found
- [ ]  In the footer, the mattermost link has `/login` that can be removed. Users will be redirected there automatically if needed
- [ ] In the footer it says 'Code and docs licenced under the GPLv2`. This is not the case, code is licenced under the AGPLv3 and the docs are licenced under the GPLv3
- [x]  The search functionality is nice!

## Install

- [ ] Minimum specs - missing Ubuntu 16.04 LTS

## FAQ

- [ ] Docs UI Kit - is this really a frequently asked question? :)
- [ ] Contribute to Docs UI Kit - the link is to `#` and also not really relevant to LibreTime
- [ ] Search in Docs layout - this is more relevant to the Docs UI FAQs than ours

**Comment by paddatrapper** [comment]
Also I didn't test the links to the docs in the actual application

**Comment by zklosko** [comment]
> Had to rename docs/gemfile to docs/Gemfile.

Did you run `bundler install` before `jekyll serve`?

> Docs UI Kit - is this really a frequently asked question? :)

Haha, you got me. That's what I get for using a template. 😄 

> The sub-title ('Everything you need to get your terrestrial/on-line...') is hard to read over the background. Maybe the font should be thicker?

I agree but I feel bad for deviating too far from what @ned-kelly created with his splash page. At this point I'm trying to make everything cohesive. The entire front page seems a bit wordy to me too. I'll take another look at it. 

I'd also love to have a one-liner on our website for install similar to other projects that would automatically unpack and install the latest stable release.

```
curl https://libretime.org/start | sudo bash
```

**Comment by paddatrapper** [comment]
> Did you run bundler install before jekyll serve?

Yes:

```
$ bundle install
Could not locate Gemfile
```

> curl https://libretime.org/start | sudo bash

Please no. At least if you download it first, you can verify that this script is actually what you want to do. Also I think that the skills required to `git clone` and `sudo ./install -fap` are the same as knowing how to open a command line and run `curl | bash`.

**Comment by zklosko** [comment]
Ah, okay.

You were in the /docs directory when you ran `bundle install`, right?

**Comment by paddatrapper** [comment]
Yeah. Renaming and running bundle install works as expected

**Comment by zklosko** [comment]
```
'no linger actively maintained' - does this need to be bold? I think the important part here is that LibreTime is open source, not that Airtime is not maintained
```

I agree. I think the whole thing needs to be rewritten. Too political.

**Comment by zklosko** [comment]
@paddatrapper I think it's closer to being done

**Comment by paddatrapper** [comment]
I am still getting the error with `gemfile` with a lowercase `g`

# Home page

- [ ] on smaller screens, the 'Photo by Leo Wieling on Unsplash' photo credit comes underneath the Key Features heading
- [ ] Footer code licence is wrong - it should be the AGPLv3
- [ ] Spelling error under Stable Releases - `Throughly` should be Thoroughly. Also the `rest(ed)?` doesn't make sense. Maybe it should be 'Thoroughly tested so that you can relax. Also release snapshots should be official releases

**Comment by zklosko** [comment]
@paddatrapper Bummer. Hopefully it's behaving now.

**Comment by paddatrapper** [comment]
The 'Thoroughly tested so you can relax?' isn't a question, so I'm not sure what the purpose of the '?' is. If you want to match the other side, maybe change it to 'Want stable and reliable?'

**Comment by zklosko** [comment]
@paddatrapper @Robbt please review. Do you think this can close #376?

**Comment by Robbt** [comment]
It probably can close #376 - I'll try to set up a testing environment today and give it a look.

**Comment by paddatrapper** [comment]
@zklosko - /docs/settings has a couple 404 on screenshots. Also the 'Improve this page' link is broken - for example https://github.com/Libretime/libretime/edit/master/docs/_docs/settings.md

**Comment by zklosko** [comment]
@paddatrapper Okay. I'll fix the screenshots. The "Improve this page" link will work once the PR is merged; we just don't have any files in the `docs/_docs` folder yet.

**Comment by Robbt** [comment]
It looks really good, big improvement over our current site which was a improvement over the first one. Seems like this more than covers all the documentation desires we had but I haven't read every page. If it works though I think it'd make sense to merge it and then fix any errors we notice. So thumbs up from my perspective. Thanks for all of the hard work on this.

**Comment by paddatrapper** [comment]
I have tried to review as many of the pages as I can. All looks good to me, so I'll merge

**Comment by narcisgarcia** [comment]
This is broken:
https://libretime.org/docs/webstreams/

**Comment by zklosko** [comment]
@narcisgarcia On which page?

**Comment by narcisgarcia** [comment]
Main menu: Webstreams
(when list is empty)

**Comment by feizing** [comment]
I was just about to submit a new issue for this, glad it was caught here.

The links with a trailing slash appear in all of these dashboard panes:
- Tracks (i.e. main dashboard pane when you click on Dashboard), 
- Playlists
- Smartblocks
- Webstreams

`Podcasts` is the only one without a trailing slash.

**Comment by paddatrapper** [comment]
@feizing are do create an issue for that
